include/swift/Basic/FileTypes.def

```diff
@@ -59,8 +59,7 @@ TYPE("raw-sib",             RawSIB,                    "sib",             "")
 TYPE("llvm-ir",             LLVM_IR,                   "ll",              "")
 TYPE("llvm-bc",             LLVM_BC,                   "bc",              "")
 TYPE("diagnostics",         SerializedDiagnostics,     "dia",             "")
-TYPE("objc-header",         ObjCHeader,                "h",               "")
-TYPE("cxx-header",          CXXHeader,                 "h",               "")
+TYPE("clang-header",        ClangHeader,               "h",               "")
```

Comment:
The name `objc-header` is significant in the output file map. I like the idea of changing this to `clang-header`, but we still have to properly consume output file maps where the key is 'objc-header'. This change is causing failures in the source compatibility suite, because the generated headers aren't being produced at all. See, e.g., https://ci.swift.org/job/swift-PR-source-compat-suite-macos/107/artifact/swift-source-compat-suite/FAIL_Starscream-examples-AutobahnTest-Autobahn.xcodeproj_5.0_BuildXcodeProjectScheme_Starscream_generic-platform-iOS.log

Reply:
I see, thank you for pointing that out

Reply:
Fix in https://github.com/apple/swift/pull/42061